bindings/rust/src/lib.rs

```diff
@@ -46,9 +46,7 @@ impl Module {
     /// Create an instance of a module.
     // TODO: support imported functions
     pub fn instantiate(self) -> Result<Instance, ()> {
-        if self.0.is_null() {
-            return Err(());
-        }
+        assert!(!self.0.is_null());
```

Comment:
Having null here is an error of using the Rust API and not a proper error emitted by the C code.